rs/cli/src/detect_neuron.rs

```diff
@@ -94,15 +94,29 @@ impl Neuron {
         Ok(neuron_id)
     }
 
-    pub async fn as_arg_vec(&self, with_auth: bool) -> anyhow::Result<Vec<String>> {
-        if !with_auth {
-            return Ok(vec![]);
-        };
-
+    pub async fn as_arg_vec(&self, require_auth: bool) -> anyhow::Result<Vec<String>> {
         // Auth required, try to find valid neuron id using HSM or with the private key
         // If private key is provided, use it without checking
-        let auth = self.get_auth().await?;
-        let neuron_id = auto_detect_neuron_id(self.network.get_nns_urls(), auth).await?;
+        let auth = match self.get_auth().await {
+            Ok(auth) => auth,
+            Err(e) => {
+                if require_auth {
+                    return Err(anyhow::anyhow!(e));
+                } else {
+                    return Ok(vec![]);
+                }
+            }
+        };
+        let neuron_id = match auto_detect_neuron_id(self.network.get_nns_urls(), auth).await {
+            Ok(neuron_id) => neuron_id,
+            Err(e) => {
+                if require_auth {
+                    return Err(anyhow::anyhow!(e));
+                } else {
+                    return Ok(vec![]);
+                }
```

Comment:
Wouldn't it make things faster if we didn't even try to do anything if we don't require auth? Just return `Ok(vec![])`

Reply:
Yes, that is one option. But then it wouldn't be easy to find out if the HSM parameters are correctly discovered. In that case, the only way to figure out if the HSM works is to try to submit the proposal.